forge-gui/res/cardsfolder/upcoming/vazi_keen_negotiator.txt

```diff
@@ -0,0 +1,15 @@
+Name:Vazi, Keen Negotiator
+ManaCost:2 B R G
+Types:Legendary Creature Human Advisor
+PT:3/3
+K:Haste
+A:AB$ Token | Cost$ T | ValidTgts$ Opponent | TokenAmount$ X | TokenScript$ c_a_treasure_sac | TokenOwner$ Targeted | SpellDescription$ Target opponent creates X Treasure tokens, where X is the number of Treasure tokens you created this turn.
+SVar:X:PlayerCountPropertyYou$TreasureTokensCreatedThisTurn
+T:Mode$ SpellCast | ValidCard$ Card | ValidActivatingPlayer$ Opponent | TriggerZones$ Battlefield | ManaFrom$ Treasure | Execute$ TrigPutCounter | TriggerDescription$ Whenever an opponent casts a spell or activates an ability, if mana from a Treasure was spent to cast or activate it, put a +1/+1 counter on target creature, then draw a card.
```

Comment:
these two trigger can be done as one, i think we have SpellAbilityCast or something